Fix minor issues in the app

droidmonkey opened the pull request:
* Fixes #11808
  * Don't show tear off menu or option to tear off if there are no tags
  * Fix "No Tags" not being shown on first hover
  * Fix issues when using a tag named "No Tags"

* Fixes https://github.com/keepassxreboot/keepassxc/issues/11662 - disable the save button when viewing Password Generator and Application Settings to restore previous behavior of toolbar

* Fixes https://github.com/keepassxreboot/keepassxc/issues/11891

* Fixes https://github.com/keepassxreboot/keepassxc/issues/11357
  * Introduces validity parameter to TOTP generator function for future use elsewhere in the code base

* Fixes #11856
  * Set the minimum recommended wordlist size to 1,296 - equal to the EFF Short List
  * Issue a clear warning when using a smaller wordlist but do not prevent generation of passphrases

* Remove start menu shortcuts on uninstall
* Fixes #12070

* Fixes https://github.com/keepassxreboot/keepassxc/issues/12153 - tags becoming unsorted in the context menu when switching between database tabs

## Testing strategy
[NOTE]: # ( Please describe in detail how you tested your changes. )
[TIP]:  # ( We expect new code to be covered by unit tests and include helpful comments. )
Tested on Windows

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ Bug fix (non-breaking change that fixes an issue)

droidmonkey committed:
Fix minor issues with tags context menu

* Fixes #11808
* Don't show tear off menu or option to tear off if there are no tags
* Fix "No Tags" not being shown on first hover
* Fix issues when using a tag named "No Tags"
* Fix #12153 - tags becoming unsorted in the context menu when switching between database tabs

droidmonkey committed:
Disable save button when viewing non-database screens

* Fixes #11662 - disable the save button when viewing Password Generator and Application Settings to restore previous behavior of toolbar

droidmonkey committed:
Show main page when editing entry or database settings

* Fixes #11891

droidmonkey committed:
Don't add space to invalid TOTP strings

* Fixes #11357
* Introduces validity parameter to TOTP generator function for future use elsewhere in the code base
* Fixes this in preview panel and TOTP dialog
* Disable actions to copy/show TOTP if the settings are invalid
* Show an error message on the TOTP setup dialog if the settings are invalid
* Show a TOTP icon with an x if the settings are invalid

xboxones1 commented:
1) During testing I noticed that the problem with TOTP is not only on the preview panel but also when viewing from the context menu TOTP -> Show TOTP

![1](https://github.com/user-attachments/assets/52040c17-f562-463d-9d05-4af2ad8d93f9)

2) If the default TOTPs are hidden in the preview panel, the progress bar is not hidden and continues to show. 

![test](https://github.com/user-attachments/assets/a3d1dc96-c4ba-403e-a5a8-48162a0852da)

droidmonkey commented:
Egad, well my fix works for the pop up panel too so will apply there. Interesting I didn't see the progress bar stay on but will add more assurance its hidden.

varjolintu commented:
I also see this weird space inside the word in macOS:
<img width="288" alt="Screenshot 2025-05-18 at 14 15 13" src="https://github.com/user-attachments/assets/156f51bf-bec7-4330-b3c4-357951a914ff" />

droidmonkey commented:
Ok got both issues fixed:

![image](https://github.com/user-attachments/assets/8b591025-5801-41da-82a7-6aaabc9c6c6c)

Also properly hiding the progress bar and text label if the TOTP code isn't valid on the popup dialog.

varjolintu commented:
Maybe the copy button should be disabled also if the TOTP is not valid? After all it's just copying the "Invalid Key" value to clipboard.

droidmonkey committed:
Fix handling of small passphrase wordlists

* Fixes #11856
* Set the minimum recommended wordlist size to 1,296 - equal to the EFF Short List
* Issue a clear warning when using a smaller wordlist but do not prevent generation of passphrases
* Improve wording when removing custom wordlist

droidmonkey commented:
Added another fix for the passphrase generator wordlist size handling:

![image](https://github.com/user-attachments/assets/3b7e79a3-c94e-4fd5-85ba-4bba2b09c70c)

I also added disabling the copy button on the TOTP dialog if the code is invalid

varjolintu reviewed (approved):
Now when the Copy button is disabled in the dialog, it would make sense also disable the Copy TOTP and Copy Password and TOTP menu items when TOTP is invalid. Not mandatory though.

xboxones1 commented:
And show qr-code can also be disabled

droidmonkey commented:
The buttons are only testing if totp is configured... that would be a bigger change to the code logic deep in the totp settings. I will consider this.

droidmonkey commented:
Ok well this turned out much easier than I expected and actually improves a lot of behaviors:

![image](https://github.com/user-attachments/assets/fa941e62-49bd-43b5-bd5d-121b695d6618)

![image](https://github.com/user-attachments/assets/0abf91a7-5acc-4cc9-9f44-9ae368ea209a)

Give this a thorough test, but I think I hit everything, even copying TOTP from the auto-type dialog and browser extension

droidmonkey committed:
Replace newlines with HTML line breaks in message dialogs

* Keeps readability of translation strings without losing line breaks due to forced rich text display

droidmonkey committed:
Remove start menu shortcuts on uninstall

droidmonkey committed:
Fix icon download dialog not appearing above windows

* Fixes #12070

dqnk commented:
Hi, I was checking out this branch for the password generation with shorter wordlists and I found a bug which does not happen in the current latest version (KeePassXC 2.7.10). If you open the password generator, set the password length to very short (e.g. 4 characters), the regenerate password button will get blocked and it will stay blocked even if you click on the passphrase tab and try to generate any (even adequately long) passphrase.

Steps to reproduce:
1. Open the password generator (the die with 3 dots in top bar)
2. Select the tab to generate a password (from the Password/Passphrase tabs in generator menu). Set password length to 3, click Advanced, select all character groups and have the "Pick characters from every group" selected. The regenerate button will become greyed out. This is intended, I'm assuming because the password is too short to fit a character from every group.
3. Go to the Passphrase tab without closing the "Advanced" menu first. The button will still be grayed out. I don't think this is intended.

(this is just a random thing I noticed on this branch, I did not review the code changes, but this does not happen in KeePassXC 2.7.10).

droidmonkey commented:
Good catch, probably just need to explicitly do an "enable" operation on the button since it used to be dynamic.

and fixed it

xboxones1 commented:
![123](https://github.com/user-attachments/assets/04e2827b-9212-4323-b908-c29cc48bc91b)

xboxones1 commented:
@droidmonkey, I checked and it still hasn't been fixed. Or don't you see what the problem is?

droidmonkey commented:
What hasn't been fixed? I could not replicate your gif

xboxones1 commented:
@droidmonkey, I outlined the steps to reproduce the issue, but you marked the message as outdated. I thought a GIF would be more helpful.
1) In the security settings, disable the option to hide TOTP in the preview panel.
2) Expand the preview panel so that it is visible after unlocking the database.
3) After unlocking, ensure that TOTP is configured in the first entrey that is opened.
4) After unlocking the database, TOTP should display in the preview panel, but it does not. You have to click twice, as shown in the GIF. This is the bug.
5) This issue occurs on Linux, but I can build on Windows to test

droidmonkey commented:
Ah, I figured it out, fix inbound
